### ISSUE TITLE
Add free text field under Invoices & Receipts setting for hosts

### ISSUE THREAD

**Issue by alanna** [opencollective/opencollective#2907]
### User Story
As a host admin, I want to add text that appears on payment receipts to meet local regulations or my organization's needs. For example, charity entities may need to include specific language or tax ID numbers in order for those receipts to be tax deductible donations to a charity. Or they might just want to put a custom thank you message.

### Solution
Add a free text field below the setting for custom title where host admins can add some more info, which is then printed on receipts that go to financial contributors.

Field name: Extra info
Description: Add any other text to appear on payment receipts, such as your organization's tax ID number, info about tax deductibility of contributions, or a custom thank you message.

### Implementation

- One new setting in the Frontend project:
https://github.com/opencollective/opencollective-frontend
- Make sure the setting is exposed in the API:
https://github.com/opencollective/opencollective-api
- Display the text in the Invoices project:
https://github.com/opencollective/opencollective-invoices

**Comment by carbonpenguin** [comment]
In the US context, this article outlines what is required in a letter from a 501c nonprofits in order for donations of in excess of $250 to be tax deductible, so it seems to make sense to have the receipts be able to include the relevant language. An article that gives the overview of that requirement is [here](https://www.thebalancesmb.com/the-link-between-donor-acknowledgement-letters-and-taxes-2501853).

**Comment by Betree** [comment]
What would be the best spot to display this in the receipt?

![image](https://user-images.githubusercontent.com/1556356/74413735-458d8f00-4e40-11ea-9b97-fd4f3e7c54df.png)

**Comment by alanna** [comment]
Any of those are fine but lets go with Yellow in case someone wants to put a lot of text

**Comment by carbonpenguin** [comment]
Agreed!

**Comment by znarf** [comment]
A $200 bounty was attached to this issue. Anyone submitting a Pull Request will be rewarded with $200 when the Pull Request is reviewed, accepted and merged. [More info](BOUNTY.md)

This bounty is currently reserved to participants of the Open Source Festival event. If not completed there, it will be later open to everyone.

**Comment by Betree** [comment]
# Specs

## Frontend

On `Host > Settings > Invoices & Receipts` (`components/edit-collective/EditHostInvoice.js`) add a new textarea field with the label `Extra info` and the description:
> Add any other text to appear on payment receipts, such as your organization's tax ID number, info about tax deductibility of contributions, or a custom thank you message.

Must save the given extra info in `collective.settings.invoice.extraInfo`.

![image](https://user-images.githubusercontent.com/1556356/75024298-72255480-5499-11ea-8bad-431db125c6aa.png)


## Invoices

Service: https://github.com/opencollective/opencollective-invoices

Add the field in the yellow zone mentioned above. Must display multi-line texts properly.

---

Please follow our [code guidelines](https://docs.opencollective.com/help/developers/best-practices-and-code-guidelines), especially:

* The strings must be internationalized. See [/help/developers/translations](https://docs.opencollective.com/help/developers/translations)
* Whenever it's possible we must use `styled-components` to write styles. See [OC Styleguide](https://opencollective-styleguide.now.sh/)
* We're getting rid of `Bootstrap`. Please don't use it for new developments.
* Icons must be imported from the [styled-icons](http://styled-icons.js.org/) library.
* Tests written with Cypress must follow our [good practices](https://docs.opencollective.com/help/developers/testing-with-cypress) conventions.

**Comment by vinuch** [comment]
hello @znarf  i would liike to work on this issue could you assign it to me

**Comment by znarf** [comment]
@vinuch You're assigned!

**Comment by vinuch** [comment]
@znarf how do i navigate to the invoice and receipts page ? 
i have created a new collective but in the settings the sidebar stops at advanced so i cant navigate to the invoice page and view my changes

**Comment by znarf** [comment]
@znarf the settings can be found in the Organization that is hosting the Collective, not the Collective itself. You need to create an organization to see that.

**Comment by vinuch** [comment]
alright i have found that i have updated the text area  having trouble saving the value though in the api

**Comment by vinuch** [comment]
hey @znarf  i have done the first part (added the text area and saved its value to _(collective.settings.invoice.extraInfo_) the last thing is to display that value in the "yellow area " on opencollective-invoices but i cant seem to reference an "extra info" property on the test invoice that displays there, i'd expect that since im just adding the extra info feature but my question is - How can i request a receipt for this organization i created so i can see if it acually displays it 

Also ill be sending in my PR to opencollective-frontend and opencollective-invoices respectively and i wont be able to refrence this issue there i hope it wont be a problem.

**Comment by joshikeno** [comment]
@vinuch when creating a PR make use of  markdown links with the full url path to this issue

**Comment by znarf** [comment]
@vinuch You need to donate to a collective hosted by the Organization. So:

1) Activate the organization as Host (if not already done)
2) Apply to the Host using a Collective you administrate
3) Donate to the Collective using any account.
4) You should get an invoice for the donation

**Comment by vinuch** [comment]
ok thanks ill get on it

**Comment by vinuch** [comment]
@znarf i have donated but when i click download invoice i get "_Error: Network request failed_"
and this in the console 

_Access to XMLHttpRequest at 'https://invoices-staging.opencollective.com/transactions/d25ef750-0cd8-4ffd-bf74-d129b8ca979b/invoice.pdf' from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: The 'Access-Control-Allow-Origin' header has a value 'https://staging.opencollective.com' that is not equal to the supplied origin._

does anyone have an idea what could be causing this ?

**Comment by znarf** [comment]
@vinuch One solution is to bypass CORS with a browser extension (for development purpose). 

The best solution I see is to run the invoice service yourself locally. See: https://github.com/opencollective/opencollective-invoices

Make sure you have the same `.env` in both `frontend` and `invoices` projects:

```
WEBSITE_URL=http://localhost:3000
INVOICES_URL=http://localhost:3002
API_KEY=09u624Pc9F47zoGLlkg1TBSbOl2ydSAq
API_URL=https://api-staging.opencollective.com
```

**Comment by vinuch** [comment]
ok i made the _`.env`_ files the same in both projects but it was still not working so i got a browser extension to bypass CORS and that did the trick

but i have another problem i cant seem to figure out how to pass the value of the input area to `opencollective-invoices` to print it on the receipt. I tried copying the exact same thing that was done for the title but i think im still missing something because it refuses to reflect on the invoice. I noticed ho it was passed as a prop as `settings.invoiceTitle` but is accessed in `opencollective-Invoice` as `invoice.title` but i really dont get how.

@znarf could you please point me to the pull request where you made the invoiceTitle editable so i can fix it for the text area i cant seem to find it anymore 

or maybe you can tell me how to pass the value cos im beat ive been on it all day. Thanks in advance.

**Comment by znarf** [comment]
@vinuch You can browse past PRs here: https://github.com/opencollective/opencollective-invoices/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed+-author%3Aapp%2Fdependabot-preview+

**Comment by znarf** [comment]
@vinuch any progress?

**Comment by vinuch** [comment]
@znarf here are links to my pull requests 
frontend - https://github.com/opencollective/opencollective-frontend/pull/3600
invoices - https://github.com/opencollective/opencollective-invoices/pull/260
api - https://github.com/opencollective/opencollective-api/pull/3431

**Comment by vinuch** [comment]
hello @Betree hope you enjoyed your weekend .

are you done reviewing the PR by chance?

**Comment by znarf** [comment]
Sorry for the delay @vinuch, we're gonna look closely at it.

**Comment by Betree** [comment]
@znarf I've given some feedback on the PRs already. The main blocker right now is https://github.com/opencollective/opencollective-api/pull/3431#pullrequestreview-375295952, I haven't tested the code yet but I don't understand how it can work by reading at it.

**Comment by Betree** [comment]
:trophy: This issue has been completed by @vinuch  

To claim the **$200** bounty, you can either:
- Submit an invoice to https://opencollective.com/engineering/expenses/new (please mention the issue number in the expense's title)
- Or ask for an Open Collective gift card of the same value (to donate to other collectives)

**Comment by vinuch** [comment]
Thanks @betree invoice submitted.

**Comment by vinuch** [comment]
Hello @betree @znarf please could one of you look into the invoice I submitted for this issue about a month ago it’s been pending since then it’s invoice #16633 submitted to open collective engineering thank you
